django/db/models/query.py

```diff
@@ -28,6 +28,10 @@
 from django.utils.functional import cached_property, partition
 from django.utils.version import get_version
 
+# The maximum number (one less than the max to be precise) of results to fetch
+# in a get() query
+MAX_GET_RESULTS = 20
```

Comment:
I would use `MAX_GET_RESULTS = 21`, after that code and comments can be simplified:
 - `The maximum number (one less than the max to be precise) of results to fetch` -> `The maximum number of results to fetch`,
 - `clone.query.set_limits(high=MAX_GET_RESULTS + 1)` -> `clone.query.set_limits(high=MAX_GET_RESULTS)`,
 - `num if num <= MAX_GET_RESULTS else ...` -> `num if num < MAX_GET_RESULTS else ...`, etc.